hibernate-core/src/main/java/org/hibernate/query/sqm/tuple/internal/AnonymousTupleType.java

```diff
@@ -121,6 +125,20 @@ private static SqmSelectableNode<?>[] extractSqmExpressibles(SqmSubQuery<?> subQ
 		return selectClause.getSelectionItems().toArray( SqmSelectableNode[]::new );
 	}
 
+	protected static List<String> extractAliases(SqmSelectQuery<?> subQuery) {
+		final SqmSelectClause selectClause = subQuery.getQueryPart()
+				.getFirstQuerySpec()
+				.getSelectClause();
+		final var aliases = new ArrayList<String>();
+		for (final var selection : selectClause.getSelections()) {
+			final var alias = selection.getAlias();
+			selection.getSelectableNode().visitSubSelectableNodes( node ->
+					aliases.add( alias == null ? node.getAlias() : alias )
+			);
+		}
+		return aliases;
+	}
```

Comment:
I'm guessing we need this to get the `SqmSelection#getAlias` value, instead of the `SqmSelectableNode`'s one. We're already iterating through selections when calling `getSelectionItems()` though, maybe it would be worth consolidating the creation of the lists within the `extractSqmExpressibles` and `createStatementTable` methods. Also, small nitpick, but I would either get both as arrays or both as lists.

Reply:
One `SqmSelection` can contain more than one `SqmSelectableNode`.  I've tried that what you are saying in first "iteration", but that caused some twenty tests to fail. See, for example, `org.hibernate.orm.test.jpa.criteria.subquery.SubqueryTests#testNoFromClauseInSubquery`.

I'll do some refactoring into new commit so, please, check what you (don't) like.